ai_flow/common/configuration/configuration.py

```diff
@@ -0,0 +1,129 @@
+        :param key: key to retrieve
+        :param fallback: default value in case the key is missing
+        :return: the value found or a default
+        """
+        val = self._config.get(str(key).lower(), fallback)
+        if val is not None:
+            return val
+        else:
+            logger.warning("key [%s] not found in config", key)
+            raise AIFlowConfigException(f"key [{key}] not found in config")
+
+    def get_str(self, key, fallback=None) -> str:
+        val = self._config.get(str(key).lower(), fallback)
+        if val is not None:
+            return expand_env_var(val)
+        else:
+            logger.warning("key [%s] not found in config", key)
+            raise AIFlowConfigException(f"key [{key}] not found in config")
+
+    def get_int(self, key, fallback=None) -> int:
+        val = self.get_str(key, fallback=fallback)
+        try:
+            return int(val)
+        except ValueError:
+            raise AIFlowConfigException(
+                f'Failed to convert value to int. Please check "{key}" key. '
+                f'Current value: "{val}".'
+            )
+
+    def get_float(self, key, fallback=None) -> float:
+        val = self.get_str(key, fallback=fallback)
+        try:
+            return float(val)
+        except ValueError:
+            raise AIFlowConfigException(
+                f'Failed to convert value to float. Please check "{key}" key. '
+                f'Current value: "{val}".'
+            )
+
+    def get_bool(self, key, fallback=False) -> bool:
+        """
+        Get the item value as a bool.
+
+        :param key: key
+        :param fallback: fallback value
+        """
+        val = self.get_str(key, fallback=fallback)
+        if val is None:
+            return False
+        if isinstance(val, bool):
+            return val
+        s = str(val).strip().lower()
+        if s not in TRUTH_TEXT and s not in FALSE_TEXT:
+            raise ValueError("Expected a valid True or False expression.")
+        return s in TRUTH_TEXT
+
+    def as_dict(self) -> dict:
+        """Return the representation as a dictionary."""
+        return deepcopy(self._config)
+
+    def __eq__(self, other):
+        """Equality operator."""
+        return self.as_dict() == Configuration(other).as_dict()
+
+    def __str__(self) -> str:
+        return str({k: v for k, v in sorted(self.as_dict().items())})
+
+
+def get_client_configuration():
+    client_config_file_name = 'aiflow_client.yaml'
+    config_path = os.path.join(get_aiflow_home(), client_config_file_name)
+    if not os.path.isfile(config_path):
+        write_default_config(client_config_file_name)
+    config = Configuration(config_path)
+    return config
+
+
+def get_server_configuration():
+    server_config_file_name = 'aiflow_server.yaml'
+    config_path = os.path.join(get_aiflow_home(), server_config_file_name)
+    if not os.path.isfile(config_path):
```

Comment:
Implement the generation configuration logic and the configuration acquisition separately